src/service/dashboards/mod.rs

```diff
@@ -149,34 +149,26 @@ pub async fn list_dashboards(
 }
 
 #[tracing::instrument]
-pub async fn get_dashboard(
-    org_id: &str,
-    dashboard_id: &str,
-    folder_id: &str,
-) -> Result<Dashboard, DashboardError> {
-    table::dashboards::get(org_id, folder_id, dashboard_id)
+pub async fn get_dashboard(org_id: &str, dashboard_id: &str) -> Result<Dashboard, DashboardError> {
+    table::dashboards::get(org_id, dashboard_id)
         .await?
         .ok_or(DashboardError::DashboardNotFound)
+        .map(|(_f, d)| d)
 }
 
 #[tracing::instrument]
-pub async fn delete_dashboard(
-    org_id: &str,
-    dashboard_id: &str,
-    folder_id: &str,
-) -> Result<(), DashboardError> {
-    match table::dashboards::get(org_id, folder_id, dashboard_id).await? {
-        Some(_) => {} // Dashboard exists. Continue with deleting.
-        None => return Err(DashboardError::DashboardNotFound),
+pub async fn delete_dashboard(org_id: &str, dashboard_id: &str) -> Result<(), DashboardError> {
+    let Some((folder, _dash)) = table::dashboards::get(org_id, dashboard_id).await? else {
+        return Err(DashboardError::DashboardNotFound);
     };
-    table::dashboards::delete(org_id, folder_id, dashboard_id).await?;
+    table::dashboards::delete(org_id, dashboard_id).await?;
```

Comment:
I think that we should have been doing these operations inside a single transaction - both before this PR and now. 

However, since we weren't doing it before, and since I would eventually like to do a bit more refactoring to parameterize the DB connection across a variety of methods (which would be necessary to perform this in a transaction), I'll leave that improvement for a future PR.